lib/linux_admin/ssh.rb

```diff
@@ -0,0 +1,102 @@
+module LinuxAdmin
+  class SSH
+    attr_reader :ip
+    attr_reader :username
+    attr_reader :private_key
+    attr_reader :agent
+
+    def initialize(ip, username, private_key)
+      @ip = ip
+      @private_key = private_key
+      @username = username
+    end
+
+    def perform_commands(commands = [], options = {})
+      result = nil
+      socket = nil
+      if options[:agent_enabled]
+        @agent = LinuxAdmin::SSHAgent.new(@private_key, :agent_socket => options[:agent_socket])
+        @agent.with_service do |socket|
+          result = if block_given?
+                     execute_commands(commands, :socket => socket, :stdin => options[:stdin], &Proc.new)
```

Comment:
&Proc.new is getting the block that is passed to 'preform_commands' and passing next, i can also do it your way if you prefer, but this is working fine also in the current state.